### PR TITLE
Add a setting (not raise log level due to keywords)

### PR DESCRIPTION
**Pull request by saledouble**

### CODE REVIEW

**Comment by saledouble** [comment]
The new data: 
[results_not_raise.tar.gz](https://github.com/ponder-lab/Logging-Level-Evolution-Plugin/files/3683003/results_not_raise.tar.gz)

File: edu.cuny.hunter.log.core/src/edu/cuny/hunter/log/core/analysis/LogAnalyzer.java [hunk 1]
```diff
@@ -68,11 +72,20 @@
...
 
 	/**
-	 * A set of keywords in log messages.
+	 * A set of keywords in log messages for lowering log levels.
+	 */
+	private final Set<String> KEYWORDS_IN_LOG_MESSAGES_FOR_LOWERING = Stream
+			.of("fail", "disable", "error", "exception", "collision", "reboot", "terminate", "throw", "should", "start",
+					"tried", "empty", "launch", "init", "does not", "doesn't", "stop", "shut", "run", "deprecate",
+					"kill", "finish", "ready", "wait", "dead", "alive")
+			.collect(Collectors.toSet());
```
Comment (khatchad) [inline edu.cuny.hunter.log.core/src/edu/cuny/hunter/log/core/analysis/LogAnalyzer.java]: This needs to be `static`.

File: edu.cuny.hunter.log.core/src/edu/cuny/hunter/log/core/analysis/LogAnalyzer.java [hunk 2]
```diff
@@ -68,11 +72,20 @@
...
+	/**
+	 * A set of keywords in log messages for raising log levels.
 	 */
-	private final Set<String> KEYWORDS_IN_LOG_MESSAGES = Stream.of("fail", "disable", "error", "exception", "collision",
-			"reboot", "terminate", "throw", "should", "start", "should", "tried", "empty", "launch", "init", "does not",
-			"doesn't", "stop", "shut", "run", "deprecate", "kill", "finish", "ready", "wait")
+	private final Set<String> KEYWORDS_IN_LOG_MESSAGES_FOR_RAISING = Stream
+			.of("fail", "disable", "error", "exception", "collision", "reboot", "terminate", "throw", "should", "start",
+					"tried", "empty", "launch", "init", "does not", "doesn't", "stop", "shut", "run", "deprecate",
+					"kill", "finish", "ready", "wait", "dead", "alive")
```
Comment (khatchad) [inline edu.cuny.hunter.log.core/src/edu/cuny/hunter/log/core/analysis/LogAnalyzer.java]: This needs to be `static`.

**Comment by saledouble** [comment]
Log transformations for `selenium` and `IRCT-API` before and after the setting we made:

[results.tar.gz](https://github.com/ponder-lab/Logging-Level-Evolution-Plugin/files/3704548/results.tar.gz)

I also tested `ReactionDecoder`, but there were no transformations, so I did not attach its results. I tried to build other small benchmarks today (`jacpFx` and `deviation-upload`), but it was not successful, so I could not collect results for them today.

**Comment by khatchad** [comment]
Were there results for ReactionDecoder beforehand?

**Comment by saledouble** [comment]
>Were there results for ReactionDecoder beforehand?

I don't think so.

**Comment by saledouble** [comment]
Data for N = 24000:
[selenium_24000_not_raise.tar.gz](https://github.com/ponder-lab/Logging-Level-Evolution-Plugin/files/3709065/selenium_24000_not_raise.tar.gz)

I don't have original data before the change now. I am going to get them.

**Comment by saledouble** [comment]
The original data: 
[selenium_24000_original.tar.gz](https://github.com/ponder-lab/Logging-Level-Evolution-Plugin/files/3709309/selenium_24000_original.tar.gz)

The transformation CSV files for comparison with marked keywords in log expression:
[transformations_24000.tar.gz](https://github.com/ponder-lab/Logging-Level-Evolution-Plugin/files/3709317/transformations_24000.tar.gz)

**Comment by saledouble** [comment]
Make sure that remove all `.git` directories that were generated by the previous version of the tool. Otherwise, you would receive an exception.

**Comment by khatchad** [comment]
> Make sure that remove all `.git` directories that were generated by the previous version of the tool. Otherwise, you would receive an error message.

What about `.git` directories that were *not* generated by the tool?

**Pull request merged**